lib/logstash/inputs/tcp.rb

```diff
@@ -144,22 +150,15 @@ def register
     if server?
       ssl_context = get_ssl_context(SslOptions)
 
-      # RubyObject#to_java is not threadsafe, and we cannot guarantee
-      # that ours is the only reference to the underlying logger, which
-      # is memoized at a class level.
-      log4j_logger = PLUGIN_GLOBAL_MUTEX.synchronize do
-        @logger.to_java(org.apache.logging.log4j.Logger)
```

Comment:
this wasn't a good idea since JRuby just translates the method calls for the interface:
```
[main][43598b08638ad7cf6d5fdd618086eb34c6ea0045bee42e8161ff2bc4eda84d67] An exception org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `isDebugEnabled' for #<LogStash::Logging::Logger:0x63d4cb1b>
was thrown by a user handler's exceptionCaught() method while handling the following exception:
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_171]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_171]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_171]
	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_171]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_171]

```